storagenode/nodestats/service.go

```diff
@@ -99,19 +112,20 @@ func (s *Service) GetDailyStorageUsedForSatellite(ctx context.Context, satellite
 	return fromSpaceUsageResponse(resp, satelliteID), nil
 }
 
-// getGRPCClientForSatellite inits GRPC client for the satellite by id
-func (s *Service) getGRPCClientForSatellite(ctx context.Context, satelliteID storj.NodeID) (pb.NodeStatsClient, error) {
+// getGRPCClientForSatellite inits GRPC client for the satellite by id.
+// Returns client and GRPC client connection in order to be closed
+func (s *Service) getGRPCClientForSatellite(ctx context.Context, satelliteID storj.NodeID) (pb.NodeStatsClient, *grpc.ClientConn, error) {
```

Comment:
Let's make separate type rather return both `pb.NodeStatsClient` and `*grpc.ClientConn`. Also rename this function to `DialNodeStats` to make it clear that this will dial.